live-tests/tests/common/mod.rs

```diff
@@ -0,0 +1,248 @@
+    internal_dns::resolver::Resolver::new_from_subnet(log.clone(), subnet)
+        .with_context(|| {
+            format!("creating DNS resolver for subnet {}", subnet.net())
+        })
+}
+
+/// Creates a DataStore pointing at the CockroachDB cluster that's in DNS
+async fn create_datastore(
+    log: &slog::Logger,
+    resolver: &Resolver,
+) -> Result<Arc<DataStore>, anyhow::Error> {
+    let sockaddrs = resolver
+        .lookup_all_socket_v6(ServiceName::Cockroach)
+        .await
+        .context("resolving CockroachDB")?;
+
+    let url = format!(
+        "postgresql://root@{}/omicron?sslmode=disable",
+        sockaddrs
+            .into_iter()
+            .map(|a| a.to_string())
+            .collect::<Vec<_>>()
+            .join(",")
+    )
+    .parse::<PostgresConfigWithUrl>()
+    .context("failed to parse constructed postgres URL")?;
+
+    let db_config = nexus_db_queries::db::Config { url };
+    let pool = Arc::new(nexus_db_queries::db::Pool::new(log, &db_config));
+    DataStore::new_failfast(log, pool)
+        .await
+        .context("creating DataStore")
+        .map(Arc::new)
+}
+
+/// Performs quick checks to determine if the user is running these tests in the
+/// wrong place and bails out if so
+///
+/// This isn't perfect but seeks to fail fast in obviously bogus environments
+/// that someone might accidentally try to run this in.
+async fn check_execution_environment(
+    resolver: &Resolver,
+) -> Result<(), anyhow::Error> {
+    ensure!(
+        cfg!(target_os = "illumos"),
+        "live tests can only be run on deployed systems, which run illumos"
+    );
+
+    // The only real requirement for these tests is that they're run from a
+    // place with connectivity to the underlay network of a deployed control
+    // plane.  The easiest way to tell is to look up something in internal DNS.
+    resolver.lookup_ip(ServiceName::InternalDns).await.map_err(|e| {
+        let text = format!(
+            "check_execution_environment(): failed to look up internal DNS \
+                 in the internal DNS servers.\n\n \
+                 Are you trying to run this in a development environment?  \
+                 This test can only be run on deployed systems and only from a \
+                 context with connectivity to the underlay network.\n\n \
+                 raw error: {}",
+            slog_error_chain::InlineErrorChain::new(&e)
+        );
+        anyhow!("{}", textwrap::wrap(&text, 80).join("\n"))
+    })?;
+
+    // Warn the user if the temporary directory is /tmp.  This check is
+    // heuristic.  There are other ways they may have specified a tmpfs
+    // temporary directory and we don't claim to catch all of them.
+    //
+    // We could also just go ahead and use /var/tmp, but it's not clear we can
+    // reliably do that at this point (if Rust or other components have cached
+    // TMPDIR) and it would be hard to override.
```

Comment:
Interesting, you could use a setup script to set `TMPDIR` outside of the test process.

Reply:
Does the test process inherit the environment of the setup script?

Reply:
Yes, if you write it out to `$NEXTEST_ENV`: https://nexte.st/docs/configuration/setup-scripts/#environment-variables

(This is how the crdb-seed setup script works)